[{"line":1,"elements":[{"start_timestamp":"2025-09-01T06:12:38.814Z","line":2,"name":"Failure while executing Cucumber","description":"","id":"failure;failure-while-executing-cucumber","type":"scenario","keyword":"Scenario","steps":[{"result":{"duration":0,"status":"passed"},"line":3,"name":"Cucumber failed while executing","match":{"arguments":[],"location":"io.cucumber.core.Failure.failure_while_executing_cucumber()"},"keyword":"When "},{"result":{"duration":0,"error_message":"io.cucumber.core.exception.CompositeCucumberException: There were 2 exceptions. The details are in the stacktrace below.\r\n\tat io.cucumber.core.runtime.RethrowingThrowableCollector.getThrowable(RethrowingThrowableCollector.java:57)\r\n\tat io.cucumber.core.runtime.CucumberExecutionContext.getThrowable(CucumberExecutionContext.java:109)\r\n\tat io.cucumber.core.runtime.CucumberExecutionContext.finishTestRun(CucumberExecutionContext.java:104)\r\n\tat io.cucumber.testng.TestNGCucumberRunner.finish(TestNGCucumberRunner.java:145)\r\n\tat io.cucumber.testng.AbstractTestNGCucumberTests.tearDownClass(AbstractTestNGCucumberTests.java:57)\r\n\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)\r\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:580)\r\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\r\n\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:71)\r\n\tat org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:400)\r\n\tat org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:333)\r\n\tat org.testng.internal.invokers.TestMethodWorker.invokeAfterClassConfigurations(TestMethodWorker.java:238)\r\n\tat org.testng.internal.invokers.TestMethodWorker.invokeAfterClassMethods(TestMethodWorker.java:223)\r\n\tat org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:137)\r\n\tat java.base/java.util.ArrayList.forEach(ArrayList.java:1596)\r\n\tat org.testng.TestRunner.privateRun(TestRunner.java:739)\r\n\tat org.testng.TestRunner.run(TestRunner.java:614)\r\n\tat org.testng.SuiteRunner.runTest(SuiteRunner.java:421)\r\n\tat org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)\r\n\tat org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)\r\n\tat org.testng.SuiteRunner.run(SuiteRunner.java:312)\r\n\tat org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)\r\n\tat org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)\r\n\tat org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)\r\n\tat org.testng.TestNG.runSuitesLocally(TestNG.java:1208)\r\n\tat org.testng.TestNG.runSuites(TestNG.java:1112)\r\n\tat org.testng.TestNG.run(TestNG.java:1079)\r\n\tat org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)\r\n\tat org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)\r\n\tat org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)\r\n\tSuppressed: io.cucumber.core.runner.DuplicateStepDefinitionException: Duplicate step definitions in stepdefinitions.FlightBookingSteps.flight_results_should_be_displayed() and stepdefinitions.FlightSearchSteps.flight_results_should_be_displayed()\r\n\t\tat io.cucumber.core.runner.CachingGlue.lambda$prepareGlue$3(CachingGlue.java:278)\r\n\t\tat java.base/java.util.ArrayList.forEach(ArrayList.java:1596)\r\n\t\tat io.cucumber.core.runner.CachingGlue.prepareGlue(CachingGlue.java:272)\r\n\t\tat io.cucumber.core.runner.Runner.runPickle(Runner.java:72)\r\n\t\tat io.cucumber.testng.TestNGCucumberRunner.lambda$runScenario$1(TestNGCucumberRunner.java:132)\r\n\t\tat io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:137)\r\n\t\tat io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)\r\n\t\tat io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:137)\r\n\t\tat io.cucumber.testng.TestNGCucumberRunner.runScenario(TestNGCucumberRunner.java:129)\r\n\t\tat io.cucumber.testng.AbstractTestNGCucumberTests.runScenario(AbstractTestNGCucumberTests.java:35)\r\n\t\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)\r\n\t\tat java.base/java.lang.reflect.Method.invoke(Method.java:580)\r\n\t\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\r\n\t\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686)\r\n\t\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)\r\n\t\tat org.testng.internal.invokers.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:81)\r\n\t\tat org.testng.internal.thread.Async$AsyncTask.run(Async.java:34)\r\n\t\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\r\n\t\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\r\n\t\tat java.base/java.lang.Thread.run(Thread.java:1583)\r\n\tSuppressed: io.cucumber.core.runner.DuplicateStepDefinitionException: Duplicate step definitions in stepdefinitions.FlightBookingSteps.flight_results_should_be_displayed() and stepdefinitions.FlightSearchSteps.flight_results_should_be_displayed()\r\n\t\tat io.cucumber.core.runner.CachingGlue.lambda$prepareGlue$3(CachingGlue.java:278)\r\n\t\tat java.base/java.util.ArrayList.forEach(ArrayList.java:1596)\r\n\t\tat io.cucumber.core.runner.CachingGlue.prepareGlue(CachingGlue.java:272)\r\n\t\tat io.cucumber.core.runner.Runner.runPickle(Runner.java:72)\r\n\t\tat io.cucumber.testng.TestNGCucumberRunner.lambda$runScenario$1(TestNGCucumberRunner.java:132)\r\n\t\tat io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:137)\r\n\t\tat io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)\r\n\t\tat io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:137)\r\n\t\tat io.cucumber.testng.TestNGCucumberRunner.runScenario(TestNGCucumberRunner.java:129)\r\n\t\tat io.cucumber.testng.AbstractTestNGCucumberTests.runScenario(AbstractTestNGCucumberTests.java:35)\r\n\t\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)\r\n\t\tat java.base/java.lang.reflect.Method.invoke(Method.java:580)\r\n\t\tat org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)\r\n\t\tat org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686)\r\n\t\tat org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)\r\n\t\tat org.testng.internal.invokers.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:81)\r\n\t\tat org.testng.internal.thread.Async$AsyncTask.run(Async.java:34)\r\n\t\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\r\n\t\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\r\n\t\tat java.base/java.lang.Thread.run(Thread.java:1583)\r\n","status":"failed"},"line":4,"name":"Cucumber will report this error:","match":{"arguments":[],"location":"io.cucumber.core.Failure.cucumber_reports_this_error()"},"keyword":"Then "}]}],"name":"Test run failed","description":"There were errors during the execution","id":"failure","keyword":"Feature","uri":"classpath:io/cucumber/core/failure.feature","tags":[]}]